mockgcp/mockcontainer/names.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockcontainer
+
+import (
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type clusterName struct {
+	Project  *projects.ProjectData
+	Location string
+	Cluster  string
+}
+
+func (n *clusterName) String() string {
+	return "projects/" + n.Project.ID + "/locations/" + n.Location + "/clusters/" + n.Cluster
+}
+
+func (n *clusterName) NodePool(nodePool string) *nodePoolName {
+	return &nodePoolName{
+		Project:  n.Project,
+		Location: n.Location,
+		Cluster:  n.Cluster,
+		NodePool: nodePool,
+	}
+}
+
+// parseClusterName parses a string into a clusterName.
+// The expected form is `projects/*/locations/*/clusters/*`.
+func (s *MockService) parseClusterName(name string) (*clusterName, error) {
+	tokens := strings.Split(name, "/")
+
+	if len(tokens) == 6 && tokens[0] == "projects" && tokens[2] == "locations" && tokens[4] == "clusters" {
+		project, err := s.projects.GetProjectByID(tokens[1])
+		if err != nil {
+			return nil, err
```

Comment:
Do we want to wrap this error in something which says could not get project from ID $tokens[1]?

Reply:
So my rule of thumb is that it's a call to our own code, I normally prefer to have it return a reasonable error message.  It knows the projectID, so I don't have any additional context to add to the error message.

In this case, it actually breaks terraform if we don't return the exact error message terraform expects, it does naive string matching (!!!), so we really don't want to wrap it here :-)